pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -820,9 +805,17 @@ func (c *Reconciler) createTaskRuns(ctx context.Context, rpt *resources.Resolved
 	ctx, span := c.tracerProvider.Tracer(TracerName).Start(ctx, "createTaskRuns")
 	defer span.End()
 	var taskRuns []*v1beta1.TaskRun
-	matrixCombinations := rpt.PipelineTask.Matrix.FanOut()
+	var matrixCombinations []v1beta1.Params
+
+	if rpt.PipelineTask.IsMatrixed() {
+		matrixCombinations = rpt.PipelineTask.Matrix.FanOut()
+	}
+
 	for i, taskRunName := range rpt.TaskRunNames {
-		params := matrixCombinations[i]
+		var params v1beta1.Params
+		if len(matrixCombinations) > 0 {
+			params = matrixCombinations[i]
+		}
```

Comment:
Just curious can we guarantee that len(matrixCombinations)==len(rpt.TaskRunNames) if it is matrx? 
Another version from chatgpt's suggestion could be:
```
	matrixCombinations := rpt.PipelineTask.Matrix.FanOut()

	for i, taskRunName := range rpt.TaskRunNames {
		params := v1beta1.Params{}
		if len(matrixCombinations) > i {
			params = matrixCombinations[i]
		}
```

Reply:
@Yongxuanzhang Yes, it should be guaranteed since the TaskRunNames are created based on the number of matrix combinations. Thanks for the suggestion!